sw/device/lib/runtime/epmp.h

```diff
@@ -0,0 +1,550 @@
+ *   * Machine Mode Whitelist Policy is enabled (mseccfg.MMWP = 1)
+ *   * Machine Mode Lockdown is disabled (mseccfg.MML = 0)
+ */
+
+/**
+ * EPMP constants.
+ */
+typedef enum {
+  kEpmpGranularity = 0,
+  kEpmpNumRegions = 16,
+} epmp_constants_t;
+
+/**
+ * EPMP entry permissions.
+ *
+ * Entries configured with locked permissions may only be modified
+ * if Rule Locking Bypass is set (mseccfg.RLB = 1).
+ *
+ * When Machine Mode Lockdown is disabled (mseccfg.MML = 0) the
+ * combination R=0 W=1 is reserved. This is the assumed state and so
+ * it is not possible to set these values. The combination R=1 W=1
+ * X=1 has also been reserved by this library and may not be used
+ *
+ * An entry may only be configured with unlocked permissions if the
+ * entry is also configured as OFF.
+ *
+ * Note: permissions may have different meanings when Machine Mode
+ * Lockdown (mseccfg.MML) is set.
+ */
+typedef enum {
+  /**
+   * Unlocked with R=0, W=0 and X=0.
+   *
+   * Note: full access (R=1, W=1 and X=1) in Machine Mode. Only use
+   * in disabled (OFF) entries.
+   */
+  kEpmpPermUnlocked,
+
+  /**
+   * Locked with R=0, W=0 and X=0.
+   */
+  kEpmpPermLockedNoAccess,
+
+  /**
+   * Locked with R=0, W=0 and X=1.
+   */
+  kEpmpPermLockedExecuteOnly,
+
+  /**
+   * Locked with R=1, W=0 and X=0.
+   */
+  kEpmpPermLockedReadOnly,
+
+  /**
+   * Locked with R=1, W=0 and X=1.
+   */
+  kEpmpPermLockedReadExecute,
+
+  /**
+   * Locked with R=1, W=1 and X=0.
+   */
+  kEpmpPermLockedReadWrite,
+} epmp_perm_t;
+
+/**
+ * EPMP region specification.
+ *
+ * Provides the start and end addresses of a particular region. These addresses
+ * are byte-aligned (i.e. they are like regular pointers rather than encoded
+ * addresses).
+ *
+ * The `start` address is inclusive and the `end` address is exclusive.
+ */
+typedef struct epmp_region {
+  uintptr_t start;
+  uintptr_t end;
+} epmp_region_t;
+
+/**
+ * EPMP entry index.
+ */
+typedef enum {
+  kEpmpEntry0,
+  kEpmpEntry1,
+  kEpmpEntry2,
+  kEpmpEntry3,
+  kEpmpEntry4,
+  kEpmpEntry5,
+  kEpmpEntry6,
+  kEpmpEntry7,
+  kEpmpEntry8,
+  kEpmpEntry9,
+  kEpmpEntry10,
+  kEpmpEntry11,
+  kEpmpEntry12,
+  kEpmpEntry13,
+  kEpmpEntry14,
+  kEpmpEntry15,
+} epmp_entry_t;
+
+/**
+ * EPMP entry count.
+ */
+typedef enum {
+  kEpmpEntryCount0,
+  kEpmpEntryCount1,
+  kEpmpEntryCount2,
+  kEpmpEntryCount3,
+  kEpmpEntryCount4,
+  kEpmpEntryCount5,
+  kEpmpEntryCount6,
+  kEpmpEntryCount7,
+  kEpmpEntryCount8,
+  kEpmpEntryCount9,
+  kEpmpEntryCount10,
+  kEpmpEntryCount11,
+  kEpmpEntryCount12,
+  kEpmpEntryCount13,
+  kEpmpEntryCount14,
+  kEpmpEntryCount15,
+  kEpmpEntryCount16,
+} epmp_entry_count_t;
+
+/**
+ * EPMP generic status codes.
+ *
+ * These error codes can be used by any function. However if a function
+ * requires additional status codes, it must define a set of status codes to
+ * be used exclusively by such function.
+ */
+typedef enum {
+  kEpmpOk = 0,
+  kEpmpError,
+  kEpmpBadArg,
```

Comment:
```suggestion
  kEpmpError = 1,
  kEpmpBadArg = 2,
```
(just for consistency).